### PR TITLE
Sp 48 virtual room settings page

### PR DESCRIPTION
**Pull request by kikkiii21**
Want to merge this user story with the other branch I am working on to complete the virtual room functionality.
Mock data was used since we are having issues with the back end. Once the back end is fixed I can update my functions. 

the settings page was implemented as it fetches study room data of the owner of the page. If you are not the owner, you are redirected to study room home page (in case you try to access typing the URL directly)
The update and delete functions are implemented but not functional as we are waiting for the backend

### CODE REVIEW

File: src/components/CustomMUIComponents/CustomButtons.js [hunk 1]
```diff
@@ -28,6 +28,19 @@ export const PrimaryButton2 = ({content, width, onClick}) => {
         </Button>
     );
 }
+export const PrimaryButton3 = ({content, width, onClick}) => {
+    return (
+        <Button
+            onClick={onClick}
+            variant="contained"
+            style={{
+                paddingTop: '10px', paddingBottom: '10px',
```
Comment (guaj) [inline src/components/CustomMUIComponents/CustomButtons.js:37]: for consistency, put one attribute per line

File: src/components/StudyRoom/StudyRoomCreate.js [hunk 2]
```diff
@@ -30,11 +31,21 @@ export default function RoomCreation() {
...
+    React.useEffect(()=> {
+        let user = localStorage.getItem("username")
+        setRoomData({...roomData, owner: user})
+        })
 
+    React.useEffect( ()=>{
+        fetchData();
+    },[])
+    function fetchData() {
+        axios.get('http://localhost:5000/friend/murth')
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomCreate.js:43]: url should be / friend / {variable}. You are hard coding the variable to get.

File: src/components/StudyRoom/StudyRoomCreate.js [hunk 3]
```diff
@@ -30,11 +31,21 @@ export default function RoomCreation() {
...
+            .then(res => {
+                console.log(res);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomCreate.js:45]: remove the console.log if the code is working

File: src/components/StudyRoom/StudyRoomCreate.js [hunk 4]
```diff
@@ -30,11 +31,21 @@ export default function RoomCreation() {
         participants:[],
     });
 
-    React.useEffect(()=>{
-      let user = localStorage.getItem("username")
-        setRoomData({...roomData, owner:user});
-    })
+    React.useEffect(()=> {
+        let user = localStorage.getItem("username")
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomCreate.js:35]: JSON.parse(localStorage.getItem("email"));

Comment (guaj) [reply src/components/StudyRoom/StudyRoomCreate.js:35]: owner is the owner email.

File: src/components/StudyRoom/StudyRoomHome.js [hunk 5]
```diff
@@ -12,33 +12,52 @@ import Card from "@mui/material/Card";
...
+    });
 
     //todo get room title,icon,description and creation date in the study room snippet
+    useEffect(() => {
+        getData();
+    },[])
+
+    function getData(){
+        let user = localStorage.getItem("username");
+        console.log(user);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomHome.js:31]: remove console.log if code is working

File: src/components/StudyRoom/StudyRoomHome.js [hunk 6]
```diff
@@ -12,33 +12,52 @@ import Card from "@mui/material/Card";
...
+        axios.get(`http://localhost:5000/room/${user}`)
+            .then(res => {
+                console.log(res.data[0]);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomHome.js:34]: same ...

File: src/components/StudyRoom/StudyRoomHome.js [hunk 7]
```diff
@@ -12,33 +12,52 @@ import Card from "@mui/material/Card";
...
+        createdOn:'',
+    });
 
     //todo get room title,icon,description and creation date in the study room snippet
+    useEffect(() => {
+        getData();
+    },[])
+
+    function getData(){
+        let user = localStorage.getItem("username");
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomHome.js:30]: users are identified by email, not username

File: src/components/StudyRoom/StudyRoomHome.js [hunk 8]
```diff
@@ -12,33 +12,52 @@ import Card from "@mui/material/Card";
...
+                const data = res.data[0];
+                setRoomData({
+                    ...roomData,
+                    title: data.title,
+                    color: data.color,
+                    description: data.description,
+                    avatarText: data.avatarText,
+                    createdOn: data.createdAt
+                })
+                console.log(roomData)
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomHome.js:44]: remove

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    const [errorMessage, setErrorMessage] = useState('')
+    const [roomData, setRoomData] = React.useState({
+        title:'',
+        color: '',
+        avatarText:'',
+        description:'',
+        participants:[],
+    });
+
+    const user = MockStudyRoom[1];
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:24]: comment out / remove if you don't use it

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+                setRoomData({...roomData, description: user.description});
+            })
+            .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    },[])
+
+    React.useEffect(()=>{
+        let user = localStorage.getItem("username")
+        setRoomData({...roomData, owner:user});
+        fetchData();
+        console.log(roomData);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:43]: remove logs!

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+                setRoomData({...roomData, title: user.title});
+                setRoomData({...roomData, avatarText: user.avatarText});
+                setRoomData({...roomData, color: user.color});
+                setRoomData({...roomData, description: user.description});
+            })
+            .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    },[])
+
+    React.useEffect(()=>{
+        let user = localStorage.getItem("username")
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:40]: email!

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+        setRoomData({...roomData, owner:user});
+        fetchData();
+        console.log(roomData);
+    },[fetchData])
+
+    // sends updated field to db
+    function handleUpdate(e){
+        e.preventDefault();
+        console.log(roomData.title);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:49]: remove console.log

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+        let avatarIconText = SetAvatarText(roomData.title);
+        console.log('avatar text:', avatarIconText);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:51]: remove console.log

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+        setRoomData({...roomData, avatarText: avatarIconText});
+        console.log(roomData);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:53]: remove console.log

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+
+        axios.post('http://localhost:5000/room/',roomData)
+            .then(res => {
+                console.log(res);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:57]: remove console.log

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+                navigate("/study-room-home");
+            })
+            .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //deletes the room in db
+    function handleDelete(e){
+
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:65]: empty function ?

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+
+    }
+    function handleTitleChange(e){
+        let avatar = SetAvatarText( e.target.value)
+        setRoomData({...roomData, title: e.target.value, avatarText:avatar});
+        console.log(roomData.title);
+    }
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:72]: put empty line between functions.

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+        console.log(roomData)
+    }
+
+    function SetAvatarText(t){
+        let finalText='';
+        let displayText = t.split(' ').map(word => word[0]).join('');
+
+        if(displayText.length >4){
+            finalText = displayText.substring(0,4);
+        }else
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:89]: missing the { } for the else. If you dont want to use them for single line condition, make it consistent (remove it from the if too)

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+                </div>
+                <Stack direction='row' spacing={7} marginTop={2}>
+                <PrimaryButton3 width={'41vw'} content="Update" onClick={handleUpdate} />
+                <PrimaryButton2 width={'41vw'} content="Delete" onClick={handleDelete}/>
+                </Stack>
+            </form>
+        </React.Fragment>
+    )
+    return(
+        updateRoom
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:175]: you dont need 3 lines to return this constant

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+                        value={roomData.description}
+                    />
+                </div>
+                <Stack direction='row' spacing={7} marginTop={2}>
+                <PrimaryButton3 width={'41vw'} content="Update" onClick={handleUpdate} />
+                <PrimaryButton2 width={'41vw'} content="Delete" onClick={handleDelete}/>
+                </Stack>
+            </form>
+        </React.Fragment>
+    )
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:173]: empty lines

**Review by guaj — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 21]
```diff
@@ -24,6 +25,7 @@ function AppRoutes() {
             <Route path="/editProfile" element={<EditProfile />} />
             <Route path="/study-room" element={<StudyRoom />} />
             <Route path="/study-room-home" element={<StudyRoomHome />} />
+            <Route path="/study-room-settings" element={<StudyRoomSettings />} />
```
[the remark this answers is not in the document]
Comment (kikkiii21) [reply src/App.js:28]: removed

File: src/components/CustomMUIComponents/CustomButtons.js (hunk 1 above)
Comment (kikkiii21) [reply src/components/CustomMUIComponents/CustomButtons.js:37]: in my IDE i see all the formatting the same was 
Screen Shot 2022-11-02 at 7 42 46 PM

Comment (kikkiii21) [reply src/components/CustomMUIComponents/CustomButtons.js:37]: I see all the buttons that way in my IDE
Screen Shot 2022-11-02 at 7 44 29 PM

Screen Shot 2022-11-02 at 7 45 43 PM

File: src/components/StudyRoom/StudyRoomCreate.js (hunk 2 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomCreate.js:43]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomCreate.js (hunk 3 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomCreate.js:45]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomCreate.js (hunk 4 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomCreate.js:35]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomHome.js (hunk 5 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomHome.js:31]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomHome.js (hunk 6 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomHome.js:34]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomHome.js (hunk 7 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomHome.js:30]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomHome.js (hunk 8 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomHome.js:44]: not in the scope of this PR

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 10 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:43]: I dont see this on the branch

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 11 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:40]: I don;t have this on the branch im going to push again

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 17 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:94]: put empty lines between all functions

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 18 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:108]: added {} for all if-else statements

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 19 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:194]: I find it more readable like that, but I put it on 1 line

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 20 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:192]: not sure where you see empty lines here

**Comment by kikkiii21** [comment]
Ive updated the file, please confirm that its good

File: src/components/StudyRoom/StudyRoomSettings.js [hunk 22]
```diff
@@ -0,0 +1,203 @@
...
+        //     .then(res => {
+        //         console.log(res);
+        //         navigate("/study-room-home");
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //deletes the room in db
+    function handleDelete(e){
+        deleteRoom();
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomSettings.js:90]: you can put the content of the function deleteRoom in here, no need to call a function in a function here

**Review by guaj — APPROVED** [review]

File: src/components/StudyRoom/StudyRoomSettings.js (hunk 22 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:90]: I was told it was better to put axios in a separate method and call that method in here

Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomSettings.js:90]: in any case ill just put the axios in there

**Pull request merged**